kroxylicious-docs/docs/_modules/record-encryption/azure-key-vault/proc-azure-key-vault-setup-microsoft-entra-managed-identity.adoc

```diff
@@ -0,0 +1,62 @@
+
+NOTE: A Managed Identity is a kind of Service Principal that is automatically managed by Azure, and doesn't require credentials to use. Once created, the process for assigning RBAC roles is largely the same.
+
+.Prerequisites
+
+* Access to the Azure CLI
+* A user with sufficient permissions to manage the hosting resource and assign RBAC roles.
+* The Azure resource (VM, etc.) intended to host Kroxylicious must already exist.
+
+.Procedure
+
+. Enable the system-assigned managed identity on the hosting resource:
++
+[source,shell]
+----
+HOST_RESOURCE_NAME="kroxylicious-host" # <1>
+HOST_RESOURCE_GROUP="my-resource-group"
+az <RESOURCE_TYPE> identity assign \ #<2>
+  --name ${HOST_RESOURCE_NAME} \
+  --resource-group ${HOST_RESOURCE_GROUP} \
+  --query "principalId" --output tsv #<3>
+----
+<1> The name of the Azure resource hosting Kroxylicious.
+<2> The command used here depends on the hosting resource type. Replace `<RESOURCE_TYPE>` with the appropriate subcommand for your resource type (e.g., `vm`, etc.)
+<3> Outputs the object identifier (Principal ID) of the managed identity being created. You will need this identifier for the next step.
+
+. Assign the built-in Azure Key Vault RBAC role to the managed identity:
++
+[source,shell]
+----
+PRINCIPAL_ID="00000000-0000-0000-0000-000000000000" # <1>
+KEY_VAULT_NAME="kroxylicious-key-vault"
+RESOURCE_GROUP="my-resource-group" # <2>
+SCOPE_ID=$(az keyvault show --name ${KEY_VAULT_NAME} --resource-group ${RESOURCE_GROUP} --query "id" --output tsv)
+az role assignment create --assignee ${PRINCIPAL_ID} --role "Key Vault Crypto Service Encryption User" --scope ${SCOPE_ID}
+----
+<1> This is the object identifier (Principal ID) we output in the previous step.
+<2> This value should be the name of the resource group in which your key vault is deployed, which may not necessarily be the same resource group as the Kroxylicious host resource.
++
+These commands assign the built-in Key Vault Crypto Service Encryption User role to the Managed Identity, using the target Key Vault as the scope. Replace `HOST_RESOURCE_NAME`, `KEY_VAULT_NAME`, and `RESOURCE_GROUP` with your actual values.
++
+NOTE: If you aren't using the built-in RBAC roles, you will need to manually assign the following privileges to the managed identity: `Microsoft.KeyVault/vaults/keys/read`, `Microsoft.KeyVault/vaults/keys/wrap/action`, and `Microsoft.KeyVault/vaults/keys/unwrap/action`. If using Managed HSM, the identity also requires `Microsoft.KeyVault/managedhsms/rng/action`.
+
+. Verification: Confirm the role has been successfully assigned.
++
+[source,shell]
+----
+az role assignment list --assignee "kroxylicious-host" --query '[].{Role:roleDefinitionName, Scope:scope}' --output tsv
```

Comment:
Using the `vm` name didn't work here:

```
az role assignment list --assignee "app-vm-redacty" --query '[].{Role:roleDefinitionName, Scope:scope}' --output tsv
Cannot find user or service principal in graph database for 'app-vm-gwt3hortdjnxe'. If the assignee is an appId, make sure the corresponding service principal is created with 'az ad sp create --id app-vm-gwt3hortdjnxe'.
```

has to work with the systemAssignedIdentity

```
az role assignment list --assignee "${PRINCIPAL_ID}" --query '[].{Role:roleDefinitionName, Scope:scope}' --scope ${SCOPE_ID}
[
  {
    "Role": "Key Vault Crypto Service Encryption User",
    "Scope": "/subscriptions/redacted/resourceGroups/my-resource-group/providers/Microsoft.KeyVault/vaults/kroxylicious-key-vault"
  }
]
```